rust/src/tensor.rs

```diff
@@ -0,0 +1,385 @@
+                };
+                Self::new(buffer, shape, strides, names)
+            }
+
+            /// Creates a new Tensor using column major memory layout
+            pub fn new_column_major(
+                buffer: Buffer,
+                shape: Option<Vec<i64>>,
+                names: Option<Vec<&'a str>>,
+            ) -> Self {
+                let strides = match &shape {
+                    None => None,
+                    Some(ref s) => Some(compute_column_major_strides::<$native_ty>(&s)),
+                };
+                Self::new(buffer, shape, strides, names)
+            }
+
+            /// The data type of the `Tensor`
+            pub fn data_type(&self) -> &DataType {
+                &self.data_type
+            }
+
+            /// The sizes of the dimensions
+            pub fn shape(&self) -> Option<&Vec<i64>> {
+                self.shape.as_ref()
+            }
+
+            /// Returns a reference to the underlying `Buffer`
+            pub fn data(&self) -> &Buffer {
+                &self.buffer
+            }
+
+            /// The number of bytes between elements in each dimension
+            pub fn strides(&self) -> Option<&Vec<i64>> {
+                self.strides.as_ref()
+            }
+
+            /// The names of the dimensions
+            pub fn names(&self) -> Option<&Vec<&'a str>> {
+                self.names.as_ref()
+            }
+
+            /// The number of dimensions
+            pub fn ndim(&self) -> i64 {
+                match &self.shape {
+                    None => 0,
+                    Some(v) => v.len() as i64,
+                }
+            }
+
+            /// The name of dimension i
+            pub fn dim_name(&self, i: i64) -> Option<&'a str> {
+                match &self.names {
+                    None => None,
+                    Some(ref names) => Some(&names[i as usize]),
+                }
+            }
+
+            /// The total number of elements in the `Tensor`
+            pub fn size(&self) -> i64 {
+                (self.buffer.len() / mem::size_of::<$native_ty>()) as i64
+            }
+
+            /// Indicates if the data is laid out contiguously in memory
+            pub fn is_contiguous(&self) -> bool {
+                self.is_row_major() || self.is_column_major()
+            }
+
+            /// Indicates if the memory layout row major
+            pub fn is_row_major(&self) -> bool {
+                match self.shape {
+                    None => false,
+                    Some(ref s) => Some(compute_row_major_strides::<$native_ty>(s)) == self.strides,
+                }
+            }
+
+            /// Indicates if the memory layout column major
+            pub fn is_column_major(&self) -> bool {
+                match self.shape {
+                    None => false,
+                    Some(ref s) => {
+                        Some(compute_column_major_strides::<$native_ty>(s)) == self.strides
+                    }
```

Comment:
Should we omit `{}` like we do in `is_row_major`?

Reply:
`rustfmt` or perhaps `cargo fmt` usually suggests to use this style.
`cargo fmt` does not complain in this case.

Reply:
Yea, we check code format in our CI, in this case it's just that the the extra 3 characters made `cargo fmt` complain about the length of the line...

Reply:
I see!